### DISCUSSION TITLE
从文件中读取 token 运行

### DISCUSSION THREAD

**Discussion by MiaowFISH** [messense/aliyundrive-webdav#272] [Q&A] [answered]
翻了翻之前的 issues 也没见相关讨论，discussion 里倒是有提到过 https://github.com/messense/aliyundrive-webdav/discussions/110#discussioncomment-1886368

就是不指定 refresh_token 时从工作目录读取，这样可以实现开机自启（？）不然还要手动填token

**Comment by MiaowFISH** [comment, marked as answer]
我已添加了 --workdir 参数，但 --refresh_token 是强制的.
```bash
VM-8-7-ubuntu:~# aliyundrive-webdav --auto-index  --workdir /etc/aliyundrive-webdav
error: The following required arguments were not provided:
    --refresh-token <REFRESH_TOKEN>

USAGE:
    aliyundrive-webdav --refresh-token <REFRESH_TOKEN> --auto-index --auth-user <AUTH_USER> --auth-password <AUTH_PASSWORD> --workdir <WORKDIR>

For more information try --help
```

**Reply by messense** [reply]
你可以认为 `--refresh-token` 是 refresh token 的种子，`--workdir` 会保存更新后的 refresh token 值，故 `--refresh-token` 参数必填，但不需要保持最新，只有在它和 workdir 中保存的 refresh token 都失效了之后才需要重新提供 `--refresh-token` 参数。

**Reply by MiaowFISH** [reply]
> 你可以认为 `--refresh-token` 是 refresh token 的种子，`--workdir` 会保存更新后的 refresh token 值，故 `--refresh-token` 参数必填，但不需要保持最新，只有在它和 workdir 中保存的 refresh token 都失效了之后才需要重新提供 `--refresh-token` 参数。

好的，明白了

**Comment by MiaowFISH** [comment]
唔，用命令行解决了

使用 `--refresh-token $(</path/to/refresh_token)` 即可

**Comment by MiaowFISH** [comment]
新的问题，配置 `systemctl service` 的 `ExecStart` 不能使用这种语法
